src/comm/TCPSocket.cc

```diff
@@ -142,3 +142,9 @@ std::string TCPSocket::print_source()
 }
 
 short TCPSocket::source_family() { return _source_family; }
+
+bool TCPSocket::is_open()
+{
+    char c;
+    return ::recv(_socket_fd, &c, 1, MSG_PEEK | MSG_DONTWAIT) != 0;
```

Comment:
Can you add a link to some documentation, if possible.

Reply:
Is it possible to add a python test ? Maybe later.

- @drewaogle has a PR for pause. The pause should be replaced by a while loop that sleeps for 100ms -> checks for cancellation -> sleeps again.
- SO you can create a python test that has two commands {pause, create a node}
- Stop the python client, reconnect and do a write transaction followed by a check if the node exists (it should not).

Reply:
We can also incorporate such timing tests `distributed` apertureDb tests I am going to write.

Reply:
I added the link and a short explanation.